File: 1.WEB_MES/backend/server-standard/src/main/kotlin/com/gmin/serverstandard/application/domain/model/Organization.kt
package com.gmin.serverstandard.application.domain.model

import java.time.LocalDateTime

/**
 * 조직 도메인 모델
 *
 * 조직의 기본 정보를 담고 있는 도메인 엔티티입니다.
 */
data class Organization(
    val organizationId: Int,
    val organizationName: String,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
    val companyId: Int?,
    val parentOrganizationId: Int?,
    val users: List<User> = emptyList()
)

File: 1.WEB_MES/frontend/MESALL.Shared/Models/Organization.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace MESALL.Shared.Models
{
    public class Organization
    {
        public int OrganizationId { get; set; }
        public string OrganizationName { get; set; }
        public string Description { get; set; }
        public int CompanyId { get; set; }
        public int? ParentOrganizationId { get; set; }
        
        [JsonIgnore]
        public bool IsRoot => ParentOrganizationId == null;
        
        [JsonIgnore]
        public List<Organization> Children { get; set; } = new List<Organization>();
        
        public List<User> Users { get; set; } = new List<User>();
        
        [JsonIgnore]
        public bool IsDeleting { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }

    public class CreateOrganizationRequest
    {
        public string OrganizationName { get; set; }
        public int CompanyId { get; set; }
        public int[] UserIds { get; set; } = Array.Empty<int>();
        public int? ParentOrganizationId { get; set; }
    }

    public class UpdateOrganizationRequest
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public int? ParentOrganizationId { get; set; }
    }
}